src/diffusers/models/cross_attention.py

```diff
@@ -291,19 +304,35 @@ class CrossAttnProcessor:
     def __call__(
         self,
         attn: CrossAttention,
-        hidden_states,
-        encoder_hidden_states=None,
-        attention_mask=None,
+        hidden_states: FloatTensor,
+        encoder_hidden_states: Optional[FloatTensor] = None,
+        attention_mask: Optional[FloatTensor] = None,
+        encoder_attention_bias: Optional[FloatTensor] = None,
```

Comment:
Hmmm, I don't fully understand why we can't just use the `attention_mask` function argument. Why do we need a new encoder_attention_bias function argument? We could just pass `encoder_attention_bias.unsqueeze(-2)` to the `attention_mask` argument no?

The code as is now is too complicated. Also the user should only know about `attention_mask` and not have to wrap the head around both attention mask and encoder attention bias I think

Reply:
`attention_mask` is a bit of a minefield at the moment, because `unet_2d_blocks` believes that it manages that parameter.

for stable-diffusion, in practice `unet_2d_blocks` doesn't currently pass any `attention_mask` down here. so it can be safely overridden with the contents of a `cross_attention_kwargs['attention_mask']`.

but I assume this doesn't hold true for all types of model?

if we co-opt the existing `attention_mask` parameter: how would you know whether the user intends it as a _self-attention_ mask, or a _cross-attention_ mask?